packages/webapp/pages/settings/job-preferences.tsx

```diff
@@ -92,25 +97,51 @@ const JobPreferencesPage = (): ReactElement => {
   const [option, setOption] = useState(null);
 
   const opts = getCandidatePreferencesOptions(user?.id);
+  const updateQuery = useUpdateQuery(opts);
 
   const { data: preferences, isPending } = useQuery(opts);
   const { mutate: updatePreferences } = useMutation({
-    ...updateCandidatePreferencesMutationOptions(useUpdateQuery(opts), () => {
+    ...updateCandidatePreferencesMutationOptions(updateQuery, () => {
       completeAction(ActionType.UserCandidatePreferencesSaved);
     }),
     onError: () => {
       displayToast('Failed to update preferences. Please try again.');
     },
   });
   const { mutate: clearResume, isPending: isClearResumePending } = useMutation({
-    ...clearResumeMutationOptions(useUpdateQuery(opts), () => {
+    ...clearResumeMutationOptions(updateQuery, () => {
       completeAction(ActionType.UserCandidatePreferencesSaved);
     }),
     onError: () => {
       displayToast('Failed to remove uploaded CV. Please try again.');
     },
   });
 
+  const { onUpload: uploadCv, isPending: isUploadCvPending } = useUploadCv({
```

Comment:
We still end up having this in a bunch of places right?
Might be worth to extract yet another thing like "UploadCVButton"?

Reply:
You're absolutely right!